cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -158,6 +160,64 @@ func (kcm KindClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
 	return nil
 }
 
+// PreflightCheck performs and pre-checks that can find issues up front that
+// would cause problems for cluster creation.
+func (kcm KindClusterManager) PreflightCheck() []error {
+	// Check presence of docker
+	cmd := exec.Command("which", "docker")
```

Comment:
This won't work on Windows.

Reply:
`docker ps` would be a good test for 2 reasons:

1. It would surface whether `docker` is installed.
2. It would surface whether the user can talk to the docker daemon (often an issue when we forget to `gpasswd -a ${USER} docker

Reply:
Good call!